march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py

```diff
@@ -100,7 +136,10 @@ def remove_setpoint(self, index):
     def save_setpoints(self, single_joint_change=True):
         self.setpoints_history.append(copy.deepcopy(self.setpoints))    # list(...) to copy instead of pointer
         if single_joint_change:
-            self.gait_generator.save_changed_joints([self])
+            try:
+                self.gait_generator.save_changed_joints([self])
+            except AttributeError:
+                rospy.logerr("AttributeError: gait_generator not set in joint trajectories")
```

Comment:
Why did you add this `try`? I see you also removed the `gait_generator` attribute from the `__init__`, so where is it set now?